From 94114c3df929f8f08f2fa88099ed49142251a0a6 Mon Sep 17 00:00:00 2001
From: Xiaoyin Liu <xiaoyinl@users.noreply.github.com>
Date: Tue, 15 Nov 2022 07:25:34 -0500
Subject: [PATCH] 4749: Remove -DSTRSAFE_LIB and STRSAFE_LIB_IMPL

Remove -DSTRSAFE_LIB and -DSTRSAFE_LIB_IMPL from C Parser
profiles and from GDT parser scripts. If they are defined, functions
like StringCbCopyW() will not be included in the archive.
---
 .../Features/Base/data/parserprofiles/VisualStudio12_32.prf   | 2 --
 .../Features/Base/data/parserprofiles/VisualStudio12_64.prf   | 2 --
 Ghidra/Features/Base/data/parserprofiles/vs12Local.prf        | 2 --
 .../Base/ghidra_scripts/CreateDefaultGDTArchivesScript.java   | 4 ----
 .../Base/ghidra_scripts/CreateExampleGDTArchiveScript.java    | 2 --
 5 files changed, 12 deletions(-)

diff --git a/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_32.prf b/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_32.prf
index b65e31847..9bcb48fcf 100644
--- a/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_32.prf
+++ b/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_32.prf
@@ -188,8 +188,6 @@ rpcproxy.h
 -D_USE_ATTRIBUTES_FOR_SAL
 -D_CRTBLD
 -D_OPENMP_NOFORCE_MANIFEST
--DSTRSAFE_LIB
--DSTRSAFE_LIB_IMPL
 -DLPSKBINFO=LPARAM
 -D_WCHAR_T_DEFINED
 -DCONST=const
diff --git a/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_64.prf b/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_64.prf
index aa7101f64..fb5678152 100644
--- a/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_64.prf
+++ b/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_64.prf
@@ -190,8 +190,6 @@ rpcproxy.h
 -D_USE_ATTRIBUTES_FOR_SAL
 -D_CRTBLD
 -D_OPENMP_NOFORCE_MANIFEST
--DSTRSAFE_LIB
--DSTRSAFE_LIB_IMPL
 -DLPSKBINFO=LPARAM
 -D_WCHAR_T_DEFINED
 -DCONST=const
diff --git a/Ghidra/Features/Base/data/parserprofiles/vs12Local.prf b/Ghidra/Features/Base/data/parserprofiles/vs12Local.prf
index 742775050..e77388f0c 100644
--- a/Ghidra/Features/Base/data/parserprofiles/vs12Local.prf
+++ b/Ghidra/Features/Base/data/parserprofiles/vs12Local.prf
@@ -236,8 +236,6 @@ wspiapi.h
 -DCRTDLL
 -D_CRTBLD
 -D_OPENMP_NOFORCE_MANIFEST
--DSTRSAFE_LIB
--DSTRSAFE_LIB_IMPL
 -DLPSKBINFO=LPARAM
 -D_WCHAR_T_DEFINED
 -DCONST=const
diff --git a/Ghidra/Features/Base/ghidra_scripts/CreateDefaultGDTArchivesScript.java b/Ghidra/Features/Base/ghidra_scripts/CreateDefaultGDTArchivesScript.java
index fe48e7c56..644d4e176 100644
--- a/Ghidra/Features/Base/ghidra_scripts/CreateDefaultGDTArchivesScript.java
+++ b/Ghidra/Features/Base/ghidra_scripts/CreateDefaultGDTArchivesScript.java
@@ -294,8 +294,6 @@ public class CreateDefaultGDTArchivesScript extends GhidraScript {
 				"-D_USE_ATTRIBUTES_FOR_SAL",
 				"-D_CRTBLD",
 				"-D_OPENMP_NOFORCE_MANIFEST",
-				"-DSTRSAFE_LIB",
-				"-DSTRSAFE_LIB_IMPL",
 				"-DLPSKBINFO=LPARAM",
 				"-D_WCHAR_T_DEFINED",
 				"-DCONST=const",
@@ -514,8 +512,6 @@ public class CreateDefaultGDTArchivesScript extends GhidraScript {
 				"-D_USE_ATTRIBUTES_FOR_SAL",
 				"-D_CRTBLD",
 				"-D_OPENMP_NOFORCE_MANIFEST",
-				"-DSTRSAFE_LIB",
-				"-DSTRSAFE_LIB_IMPL",
 				"-DLPSKBINFO=LPARAM",
 				"-D_WCHAR_T_DEFINED",
 				"-DCONST=const",
diff --git a/Ghidra/Features/Base/ghidra_scripts/CreateExampleGDTArchiveScript.java b/Ghidra/Features/Base/ghidra_scripts/CreateExampleGDTArchiveScript.java
index 3b9397a38..1f0a97bcd 100644
--- a/Ghidra/Features/Base/ghidra_scripts/CreateExampleGDTArchiveScript.java
+++ b/Ghidra/Features/Base/ghidra_scripts/CreateExampleGDTArchiveScript.java
@@ -470,8 +470,6 @@ public class CreateExampleGDTArchiveScript extends GhidraScript {
 				"-D_USE_ATTRIBUTES_FOR_SAL",
 				"-D_CRTBLD",
 				"-D_OPENMP_NOFORCE_MANIFEST",
-				"-DSTRSAFE_LIB",
-				"-DSTRSAFE_LIB_IMPL",
 				"-DLPSKBINFO=LPARAM",
 				"-D_WCHAR_T_DEFINED",
 				"-DCONST=const",
-- 
2.38.1

